src/test/java/uk/gov/justice/hmpps/probationteams/repository/LocalDeliveryUnitRepositoryTest.kt

```diff
@@ -88,31 +81,23 @@ class LocalDeliveryUnitRepositoryTest(
 
         TestTransaction.start()
 
-        val persistentLduOpt = repository
-                .findByProbationAreaCodeAndLocalDeliveryUnitCode("ABC", lduCode)
-
-        assertThat(persistentLduOpt).isPresent
-
-        persistentLduOpt.ifPresent { persistentLdu ->
-            persistentLdu.probationTeams.remove("T1")
-            persistentLdu.probationTeams["T2"] = ProbationTeam("zzz@zzz.com")
-        }
+        assertThat(repository.findByProbationAreaCodeAndLocalDeliveryUnitCode("ABC", lduCode))
+                .hasValueSatisfying { persistentLdu ->
+                    persistentLdu.probationTeams.remove("T1")
+                    persistentLdu.probationTeams["T2"] = ProbationTeam("zzz@zzz.com")
+                }
 
         TestTransaction.flagForCommit()
         TestTransaction.end()
 
         TestTransaction.start()
 
-        val updatedPersistentLduOpt = repository
-                .findByProbationAreaCodeAndLocalDeliveryUnitCode("ABC", lduCode)
-
-        assertThat(updatedPersistentLduOpt).isPresent
-
-        updatedPersistentLduOpt.ifPresent { persistentLdu ->
-            assertThat(persistentLdu.id).isNotNull()
-            assertThat(persistentLdu.probationTeams).isEqualTo(mutableMapOf("T2" to ProbationTeam("zzz@zzz.com")))
-            assertThat(probationTeamCount(persistentLdu.id)).isEqualTo(1)
-        }
+        assertThat(repository.findByProbationAreaCodeAndLocalDeliveryUnitCode("ABC", lduCode))
+                .hasValueSatisfying { persistentLdu ->
+                    assertThat(persistentLdu.id).isNotNull()
+                    assertThat(persistentLdu.probationTeams).isEqualTo(mutableMapOf("T2" to ProbationTeam("zzz@zzz.com")))
```

Comment:
very minor but maybe `containsOnly` rather than `isEqualTo`

Reply:
`"T2" to ProbationTeam("zzz@zzz.com")` is a` Pair,`, `containsOnly` expects `Map.Entry` and I can't find a simple way of creating one.   Looks like I can change `mutableMapOf` to `mapOf` though...